test/functional/ui/syntax_conceal_spec.lua

```diff
@@ -913,4 +913,46 @@ describe('Screen', function()
     ]]}
     eq(grid_lines, {{2, 0, {{'c', 0, 3}}}})
   end)
+
+  -- Copy of Test_cursor_column_in_concealed_line_after_window_scroll in
+  -- test/functional/ui/syntax_conceal_spec.lua.
+  describe('concealed line after window scroll', function()
+    after_each(function()
+      command(':qall!')
+      os.remove('Xcolesearch')
+    end)
+
+    it('has the correct cursor column', function()
+      insert([[
+      3split
+      let m = matchadd('Conceal', '=')
+      setl conceallevel=2 concealcursor=nc
+      normal gg
+      "==expr==
+      ]])
+
+      command('write Xcolesearch')
+      feed(":so %<CR>")
+
+      -- Jump to something that is beyond the bottom of the window,
+      -- so there's a scroll down.
+      feed("/expr<CR>")
+
+      -- Are the concealed parts of the current line really hidden?
+      -- Is the window's cursor column properly updated for hidden
+      -- parts of the current line?
+      screen:expect{grid=[[
+        setl conceallevel2 concealcursornc                   |
+        normal gg                                            |
+        "{5:^expr}                                                |
```

Comment:
👍  The `^` here indicates that cursor is in column 2, which is what is expected: https://github.com/janlazo/neovim/commit/910bbc3cca796f7fa941e0f6176cd0061de0e01c#diff-183e586cc07f732cf6aaf830dd0bc468dda6e77b099af6d5b341e88d2ea3a86aR310